.github/workflows/testing-macos.yml

```diff
@@ -0,0 +1,57 @@
+name: build and test on macOS
+
+on:
+  push:
+    branches: [ master, devel ]
+  pull_request:
+    branches: [ master, devel ]
+
+jobs:
+  debug:
+
+    strategy:
+      matrix:
+        os: [macos-14, macos-13, macos-12]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - run: brew install bash gnu-time bats-core
+
+      - run: echo "$(brew --prefix gnu-time)/libexec/gnubin" >> "$GITHUB_PATH"
+
+      - run: cmake -S . -B build -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_COLOR_MAKEFILE=ON
+
+      - run: cmake --build build --target test_all benchmark
```

Comment:
It is not necessary to run benchmark on all platforms. The same grammars are tested with `test_all`, and I plan to add a separate (and faster) test in the future to make sure the feature works.